commands/FQL10Commands.js

```diff
@@ -84,19 +91,47 @@ class FQL10Commands {
    *        E.g. {
    *          functions: {
    *            MyFunction: { body: "x => x + 1" }
+   *          },
+   *          collections: {
+   *            MyCollection: {}
    *          }
    *        }
    * @returns An an object containing arrays of resource definitions by resource type.
    *          E.g. {
-   *            functions: [ { name: "MyFunction", body: "x => x + 1" }]
+   *            functions: [ { name: "MyFunction", body: "x => x + 1" }],
+   *            collections: [ { name: "MyCollection" }],
    *          }
    *
    */
-  adapt({ functions = {} }) {
-    return {
-      functions: Object.entries(functions).map(([k, v]) => {
+  adapt({ collections = {}, functions = {} }) {
+    const toArray = (obj) =>
+      Object.entries(obj).map(([k, v]) => {
         return { name: k, ...v, data: this.mergeMetadata(v.data) };
+      });
+
+    return {
+      collections: toArray(collections).map((c) => {
+        c.indexes = c.indexes ?? {};
+        c.constraints = c.constraints ?? [];
+        Object.entries(c.indexes).forEach(([idxName, idxDef]) => {
+          // Default index value order to asc to make diffing easier
+          if (idxDef.values != null) {
+            c.indexes[idxName].values = idxDef.values.map((iv) => {
+              if (iv.order == null) {
+                return { order: "asc", ...iv };
+              } else {
+                return iv;
+              }
+            });
+          }
+
+          // Default queryable to true to make diffing easier
+          idxDef.queryable = idxDef.queryable ?? true;
```

Comment:
I'm not sure this is the right behavior. This will cause indexes that require an async build to appear incomplete while they're building.

I agree with the nice-to-have with diffing, so could that be hacked in later when printing the diff or something?

Reply:
Got it. I’ll rework this